### ISSUE TITLE
Document architecture (how it works)

### ISSUE THREAD

**Issue by FeodorFitsner** [flet-dev/flet#55]
+ diagram

**Comment by jamesdbrock** [comment]
Yes please. I’m trying to understand this. If I run a native Windows Flet application, how many processes are there, and how many layers of communication? Is there a Dart-based process communicating with a Python process? Is there a Dart webserver communicating by Websockets to an Electron GUI?

**Comment by jamesdbrock** [comment]
Oh there's some good information in here. https://github.com/flet-dev/flet/blob/main/docs/flet.md

> Flet hides that complexity by replacing Flutter's declarative model (re-build a UI tree on app state changes) with an imperative (modify UI tree as program progresses).

Really? That's too bad.

**Comment by jamesdbrock** [comment]
Ok Flutter is an alternative to Webkit. So there is no Webkit involved here. It looks like there is a Flutter Dart process with bindings to the [C++ Flutter engine](https://github.com/flutter/engine), and then a separate Flet Python process which communicates with Flutter Dart through [JSON and sockets](https://github.com/flet-dev/flet/tree/main/package/lib/src)? And I guess that the subject of that communication is [UI events and VDOM diffs](https://github.com/flet-dev/flet/tree/main/package/lib/src/protocol)?

I was hoping that Flet has [direct Python bindings](https://github.com/pybind/pybind11) to the [C++ Flutter engine](https://github.com/flutter/engine)? No?

**Comment by jamesdbrock** [comment]
https://flet.dev/blog/flet-mobile-update/#flet-new-desktop-architecture

**Comment by jamesdbrock** [comment]
https://github.com/flet-dev/flet/blob/main/docs/flet.md

> Flet hides that complexity by replacing Flutter's declarative model (re-build a UI tree on app state changes) with an imperative (modify UI tree as program progresses).

Flet hides the simplicity of [Flutter’s declarative model](https://docs.flutter.dev/resources/architectural-overview#reactive-user-interfaces) by replacing it with complex imperative modifications to the UI tree?

**Comment by ndonkoHenri** [comment]
The flutter guys explained it pretty well. For them, having to modify the states by oneself could sometimes be problematic. I find this true in Flet's case.

@FeodorFitsner, you master Flutter and Flet architectures, what can you say on this today please? 
And also, why do you name it as 'hiding complexity'?
And by the way, could it be easy for Flet to switch on to the imperative style ?

The language runtime is also to be taken into consideration:
> This approach relies on certain characteristics of a language runtime (in particular, fast object instantiation and deletion). Fortunately, Dart is particularly well suited for this task.

**Comment by ypankovych** [comment]
@FeodorFitsner any updates here? Can we have detailed "under the hood" docs?
